src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubBuildStatusNotification.java

```diff
@@ -73,6 +74,9 @@ private static void createBuildCommitStatus(Run<?, ?> build, TaskListener listen
         SCMSource src = SCMSource.SourceByItem.findSource(build.getParent());
         SCMRevision revision = src != null ? SCMRevisionAction.getRevision(src, build) : null;
         if (revision != null) { // only notify if we have a revision to notify
+            GitHubEnvAction gitHubEnvAction = getGitHubEnvAction(revision);
```

Comment:
Unnecessary to add this action.

https://github.com/jenkinsci/branch-api-plugin/blob/b875b2641d60e39f2b6074838f034a1b921269f6/src/main/java/jenkins/branch/BranchNameContributor.java#L55-L58

Followed by a call to https://github.com/jenkinsci/branch-api-plugin/blob/0e8545955a5ded2f482f77055ab064786267c710/src/main/java/jenkins/branch/BranchProjectFactory.java#L162 will get you the revision and then you can use the exact same logic for testing the instance types to extract the information you need.

A single class extension plugin with just an `EnvironmentContributor` implementation will do exactly what you are attempting without requiring changes to the core plugin

Reply:
Will take a look thanks